src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -84,590 +57,29 @@ private static IArraySortHelper<T> CreateArraySortHelper()
             return defaultArraySortHelper;
         }
 
-        #region IArraySortHelper<T> Members
-
-        public void Sort(T[] keys, int index, int length, IComparer<T> comparer)
```

Comment:
Note: In corert this Sort method and the BinarySearch method below are static but here it has to be non static because it implements IArraySortHelper interface.

Perhaps in the shared file you can have it this way: 

```
        public 
#if CORECLR
        static
#endif
        void Sort(T[] keys, int index, int length, IComparer<T> comparer)
```

and 
```
        public 
#if CORECLR
        static
#endif
        int BinarySearch(T[] array, int index, int length, T value, IComparer<T> comparer)
```

Reply:
The instance methods are implemented as explicit interface implementation. I could remove them if you prefer if-defs instead but I thought the inliner will be able to remove them anyway.

Reply:
@jkotas what do you recommend?

Reply:
Having the non-optimized static Sort method is a bug prone pattern - it is easy to call the non-optimized method accidentally instead of going through the optimized instance method. Make it always an instance method so that you have to always go through `Default.` ?

Reply:
There is no non-optimized version. There is only one Sort version and the instance method exists for CoreCLR only to allow call the static method via an interface.

See https://github.com/dotnet/coreclr/pull/18125/files#diff-9bd6e2286ab776f741db7e9c3feca9e7R49

We could change all code to call via .Default and have the Default property implementation to do cached "new this" it just feels wrong for everyone to pay the price

Reply:
> instance method exists for CoreCLR only to allow call the static method via an interface

Right. All places should be calling the method via interface (in CoreCLR at least) to get the optimized version for constrained T. How do we ensure that all places in CoreCLR are calling the method via interface to get the optimized version when there is a static method as well?

Reply:
Changed it back to instance methods only